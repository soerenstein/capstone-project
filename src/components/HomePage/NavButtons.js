import React from 'react'
import styled from 'styled-components/macro'
import { ReactComponent as UserProfile } from '../Design/img/userHomepage.svg'
import { ReactComponent as UserList } from '../Design/img/userListHomepage.svg'
import { ReactComponent as AddUser } from '../Design/img/userPlusHomepage.svg'
import { ReactComponent as FavoriteUser } from '../Design/img/FavoriteUserHomepage.svg'
import { NavLink } from 'react-router-dom'

export default function NavButtons() {
  return (
    <>
      <NavWrapper>
        <NavLink to="/user-profile">
          <NavButtonWrapper>
            <UserProfile />
            <H2Styled>Profil</H2Styled>
          </NavButtonWrapper>
        </NavLink>
        <NavLink to="/create">
          <NavButtonWrapper>
            <AddUser />
            <H2Styled>Kontakt hinzuf√ºgen</H2Styled>
          </NavButtonWrapper>
        </NavLink>
        <NavLink to="/list">
          <NavButtonWrapper>
            <UserList />
            <H2Styled>Kontaktliste</H2Styled>
          </NavButtonWrapper>
        </NavLink>
        <NavLink to="/favorites">
          <NavButtonWrapper>
            <FavoriteUser />
            <H2Styled>Favoriten</H2Styled>
          </NavButtonWrapper>
        </NavLink>
      </NavWrapper>
    </>
  )
}

const NavWrapper = styled.div`
  position: absolute;
  bottom: 50px;
  display: flex;

  width: 100%;
  flex-wrap: wrap;
  justify-content: space-evenly;
  margin-top: 20px;
  & a {
    text-decoration: none;
  }
  & svg {
    margin-top: 20%;
  }
`

const NavButtonWrapper = styled.div`
  width: 22vh;
  height: 22vh;
  display: grid;
  grid-template-rows: 60% 40%;
  justify-items: center;
  margin: 10px;
  align-content: center;
  text-align: center;
  background: transparent 0% 0% no-repeat padding-box;
  box-shadow: 0px 3px 10px #00000029;
  border-radius: 30px;
`

const H2Styled = styled.h2`
  font-size: 1.1em;
  vertical-align: bottom;
  color: var(--black);
  font-weight: 500;
  margin-top: 0;
  margin-bottom: 0;
`
