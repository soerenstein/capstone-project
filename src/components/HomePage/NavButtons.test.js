import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import NavButtons from './NavButtons'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

describe('The Navigation', () => {
    it('renders the buttons', () => {
        const { getByText } = render(
            <MemoryRouter>
                <NavButtons />
            </MemoryRouter>
        )
        expect(getByText('Profil')).toBeInTheDocument()
        expect(getByText('Kontakt hinzufÃ¼gen')).toBeInTheDocument()
        expect(getByText('Kontaktliste')).toBeInTheDocument()
        expect(getByText('Favoriten')).toBeInTheDocument()
    })
  it('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <NavButtons />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
