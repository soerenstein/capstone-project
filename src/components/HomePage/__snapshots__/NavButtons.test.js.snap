// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Navigation renders correctly 1`] = `
<div
  className="NavButtons__NavWrapper-be11z7-0 bkLDAC"
>
  <a
    aria-current={null}
    href="/user-profile"
    onClick={[Function]}
  >
    <div
      className="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
    >
      <svg>
        userHomepage.svg
      </svg>
      <h2
        className="NavButtons__H2Styled-be11z7-2 hjdpHL"
      >
        Profil
      </h2>
    </div>
  </a>
  <a
    aria-current={null}
    href="/create"
    onClick={[Function]}
  >
    <div
      className="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
    >
      <svg>
        userPlusHomepage.svg
      </svg>
      <h2
        className="NavButtons__H2Styled-be11z7-2 hjdpHL"
      >
        Kontakt hinzuf√ºgen
      </h2>
    </div>
  </a>
  <a
    aria-current={null}
    href="/list"
    onClick={[Function]}
  >
    <div
      className="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
    >
      <svg>
        userListHomepage.svg
      </svg>
      <h2
        className="NavButtons__H2Styled-be11z7-2 hjdpHL"
      >
        Kontaktliste
      </h2>
    </div>
  </a>
  <a
    aria-current={null}
    href="/favorites"
    onClick={[Function]}
  >
    <div
      className="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
    >
      <svg>
        FavoriteUserHomepage.svg
      </svg>
      <h2
        className="NavButtons__H2Styled-be11z7-2 hjdpHL"
      >
        Favoriten
      </h2>
    </div>
  </a>
</div>
`;
